source/stdcpp/test/string.d

```diff
@@ -23,5 +23,20 @@ unittest
     a.push_back('a');
     assert(a.size() == 6);
     // verifying small string optimization, this is 15 on GCC, 22-23 on clang
-    assert(a.capacity == stringCapacity("hello"));
+    assert(a.capacity == stringCapacity("helloa"));
+    assert(a.front() == 'h');
+    assert(a.back() == 'a');
+    a.resize(4); // shrinks a to "hell"
+    assert(a.size() == 4);
+    auto b = std_string("Hi, this is a test for string capacity growth for a length more than the base SSO");
+    assert(b.capacity == stringCapacity("Hi, this is a test for string capacity growth for a length more than the base SSO"));
+    a.swap(b); // a and b swaps
+    assert(a.capacity == stringCapacity("Hi, this is a test for string capacity growth for a length more than the base SSO"));
+    assert(b.capacity == stringCapacity("hell")); // a was shrinked to hell so b contains 'hell'
+    b.pop_back();
+    assert(b.size() == 3);
+    assert(a.empty == 0);
+    a.clear();
+    assert(a.empty == 1);
+
```

Comment:
```suggestion
```

Reply:
why? Unecessary?

Reply:
Yes, why would you have trailing whitespace ?